### DISCUSSION TITLE
Modal route that appears over whatever current page/route is?

### DISCUSSION THREAD

**Discussion by treasurely** [remix-run/remix#4693] [Q&A]
Has anyone implemented a modal that (1) appears over whatever content is currently rendered and (2) uses a simple route like url.com/add?  I'm building a simple movie tracker app and am trying to use an "Add" button in the Navbar that renders a Modal with a form inside on top of the current page to add a new movie from wherever the user is in the app. 

If an app is structured like this, where should this modal route go so that it would appear over both the whatToWatchNext or whatIveWatched page when it's clicked?

```app 
|---root.tsx
|---routes
         |---whatToWatchNext.tsx
         |---whatIveWatched.tsx
         |---index.tsx
```

Ultimately looking to implement something similar to Twitter, which renders a modal above whatever the current rendered page is with the url twitter.com**/compose/tweet**.  Thanks in advance to anyone who has a thought on this.

**Comment by davidhousedev** [comment]
Do you want the modal to behave differently depending on the nested route. For example, do you want the modal to add a movie when the movie route is rendered, or set a movie as watched when a different route is rendered? If so, it sounds like you want the modal to use a relative route, e.g. `action=add`, then create different `add.tsx` action handlers on each of your child routes.

If the modal does the same thing each time, regardless of the child page rendered, then you'll want to use an absolute path for your form action, e.g. `action=/add`.

There's a similar dynamic with your text content. Do you want the modal to render with the same content on all child pages, or does it look the same on all child pages? If it looks the same, then you can implement the modal in a shared route component across all child routes, e.g. `root.tsx`. If it needs to look different, you may want to look at using React Portals which allow you to render something in a parent component from one of its children.

Hope this helps! Let me know if you have any more questions.

**Reply by treasurely** [reply]
Thanks!  I think absolute path is what I'm looking for -- the modal I'm trying to use would be the same regardless of route anywhere in the app & would have a select list of the media type in the modal itself (i.e. so adding a movie & book are done in the same form/modal, the user chooses which media type they're logging). 

 I'm trying to use it as a route because it has a few fields/comboboxes using data from an API & ideally I'd have the "Add" button on the Navbar wrapped in a prefetch Link.  After doing some more digging, it'd be the equivalent to [Next's "Intercepting Routes"](https://nextjs.org/blog/layouts-rfc#intercepting-routes).

**Reply by davidhousedev** [reply]
I see! Yeah, I can see how Next's intercepting routes feature would be a good fit because it references a parent route rather than recreating it.

I wonder if you could build intercepting routes "into" Remix by creating the duplicated child route files (as you would with Next) then importing a loader and action from a parent. Something like this:

```
export { loader, action } from '~/routes/add.tsx'
```

Then implement the React component itself in a parent route, and have it render as open when something from your loader is present e.g. `addMovieModalIsOpen`.

That said, this feels fairly brittle since it tightly couples parent component DOM to available child routes. Have you considered using `useFetcher` to handle the mutation and maybe using a loader without a URL segment to ensure the data is always available?

I'm out of my depth here with Remix hackery so someone else might have a better suggestion!
